icepyx/core/variables.py

```diff
@@ -446,53 +466,20 @@ def append(self, defaults=False, var_list=None, beam_list=None, keyword_list=Non
             and keyword_list == None
         ), "You must enter parameters to add to a variable subset list. If you do not want to subset by variable, ensure your is2.subsetparams dictionary does not contain the key 'Coverage'."
 
-        req_vars = {}
+        final_vars = {}
 
-        # if not hasattr(self, 'avail') or self.avail==None: self.get_avail()
-        # vgrp, paths = self.parse_var_list(self.avail)
-        # allpaths = []
-        # [allpaths.extend(np.unique(np.array(paths[p]))) for p in range(len(paths))]
         vgrp, allpaths = self.avail(options=True, internal=True)
-
         self._check_valid_lists(vgrp, allpaths, var_list, beam_list, keyword_list)
 
-        # add the mandatory variables to the data object
-        if self._vartype == "order":
-            nec_varlist = [
-                "sc_orient",
-                "sc_orient_time",
-                "atlas_sdp_gps_epoch",
-                "data_start_utc",
-                "data_end_utc",
-                "granule_start_utc",
-                "granule_end_utc",
-                "start_delta_time",
-                "end_delta_time",
-            ]
-        elif self._vartype == "file":
-            nec_varlist = [
-                "sc_orient",
-                "atlas_sdp_gps_epoch",
-                "cycle_number",
-                "rgt",
-                "data_start_utc",
-                "data_end_utc",
-            ]
-
-        # Adjust the nec_varlist for individual products
-        if self.product == "ATL11":
-            nec_varlist.remove("sc_orient")
-
         try:
-            self._check_valid_lists(vgrp, allpaths, var_list=nec_varlist)
+            self._check_valid_lists(vgrp, allpaths)
         except ValueError:
             # Assume gridded product since user input lists were previously validated
             nec_varlist = []
```

Comment:
That makes sense, I took this try/except out. 
A related thing I have been worried about is how this code will translate to a gridded data product. It seems from this try/except that we don't actually want to ever append required variables if the product is gridded. Is that correct? If that is the case then maybe we should be writing an if statement using the product into the Read and Query classes to skip required variable appending for gridded data products.

Reply:
Outcome of this comment:
Yes, we want to skip adding additional variables for gridded data products. We will do that:
1. **Read**: TBD
2. **Query**: Probably in `subsetparams`, the same place we add the new vars to Coverage

Reply:
Not yet thought through idea: could we add some sort of `required=__` where __ = 'read' or 'query' (or something) as an argument to `vars.append()`? It would ultimately behave within append the same way `defaults` does, and there could be a function in `is2ref.py` that then gets the right default list for the required+product (analogous to `is2ref._default_varlists(product)`). It farms the work out of read or query directly but capitalizes on related infrastructure and potentially allows for more future flexibility. Worth exploring?

Reply:
An update to this comment is documented in Issue #467. @JessicaS11 let me know what you think of the proposed path forward in that issue.

Reply:
@JessicaS11 I like the idea that maybe these lists of variables might be stored elsewhere. If we were to pursue it I would be tempted to make required a boolean, where the user gets to pick if they want to include the required variables or not.
However it is setup, I think an argument makes the most sense if it turns out that there is a lot of variability in which products need what variables. From #467 it looks like the groups are 1) ATL14, ATL15, ATL23 which don't have any of the required variables 2) ATL11 which has all but two 3) all the rest, which have all those variables. My leaning is that this is a small enough number of groups that we just add an additional if statement to `Read.load`. What do you think?